### PR TITLE
fix: Fix issue with key based credentials not being read correctly

### PR DESCRIPTION
**Pull request by Joffcom**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

**Review by netroy — COMMENTED** [review]
Perhaps these should be separate PRs, so that these fixes show up as individual entries in the diff and in the changelog.

**Review by maspio — DISMISSED** [review]
Ok looks good to me.

**Comment by WebWeave** [comment]
Hi, we are really looking forward to this release 😄 
Is there any ETA?

**Comment by Joffcom** [comment]
Hey @WebWeave,

I would say maybe 45 minutes until it is merged 🙂

**Review by maspio — DISMISSED** [review]

File: packages/nodes-base/utils/utilities.ts [hunk 1]
```diff
@@ -217,6 +217,34 @@ export const keysToLowercase = <T>(headers: T) => {
 	}, {} as IDataObject);
 };
 
+/**
+ * Formats a private key by removing unnecessary whitespace and adding line breaks.
+ * @param privateKey - The private key to format.
+ * @returns The formatted private key.
+ */
+export function formatPrivateKey(privateKey: string): string {
```
Comment (netroy) [inline packages/nodes-base/utils/utilities.ts:225]: In general we should be moving code out of this file, and into `this.helpers`. That way these utility functions are also available 3rd-party nodes.

Comment (Joffcom) [reply packages/nodes-base/utils/utilities.ts:225]: @netroy that isn't a bad idea, Fancy making a tech debt ticket for it and we can move them all out on our next node team tech debt day

**Review by maspio — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.2.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.2.1)

**Pull request merged**